src/EnergyPlus/PoweredInductionUnits.cc

```diff
@@ -2069,6 +2115,418 @@ void CalcParallelPIU(EnergyPlusData &state,
+        fanSignal = (1.0 - thisPIU.MinFanTurnDownRatio) * 0.5 + thisPIU.MinFanTurnDownRatio; // average speed as the initial value
+        auto f = [&state, PIUNum, zoneLoad, ZoneNode, primaryAirMassFlow](Real64 const fanSignal) {
+            return CalcVariableSpeedPIUHeatingResidual(state, fanSignal, PIUNum, zoneLoad, ZoneNode, primaryAirMassFlow, false, fanSignal);
+        };
+        General::SolveRoot(state, Acc, MaxIte, SolFla, fanSignal, f, thisPIU.MinFanTurnDownRatio, 1.0);
+        if (SolFla == -1) {
+            ShowSevereError(state, "Iteration limit exceeded in calculating variable speed fan powered box 1st stage heating fan speed");
+            ShowContinueErrorTimeStamp(state, "");
+            ShowFatalError(state, format("Series PIU control failed for {}:{} ", thisPIU.UnitType, thisPIU.Name));
+        } else if (SolFla == -2) {
+            ShowSevereError(state, "Bad starting values in calculating variable speed fan powered box 1st stage heating fan speed");
+            ShowContinueErrorTimeStamp(state, "");
+            ShowFatalError(state, format("Series PIU control failed for {}:{}", thisPIU.UnitType, thisPIU.Name));
+        } else {
+            if (thisPIU.UnitType == "AirTerminal:SingleDuct:SeriesPIU:Reheat") {
+                thisPIU.SecAirMassFlow = max(0.0, fanSignal * thisPIU.MaxTotAirMassFlow - primaryAirMassFlow);
+            } else if (thisPIU.UnitType == "AirTerminal:SingleDuct:ParallelPIU:Reheat") {
+                thisPIU.SecAirMassFlow = max(0.0, fanSignal * thisPIU.MaxSecAirMassFlow);
+            }
+        }
+    } else if (qdotDelivered1stStageMaxFan < zoneLoad) {
+        thisPIU.heatingOperatingMode = HeatOpModeType::StagedHeatSecondStage;
+        if (thisPIU.UnitType == "AirTerminal:SingleDuct:SeriesPIU:Reheat") {
+            thisPIU.SecAirMassFlow = max(0.0, thisPIU.MaxTotAirMassFlow - primaryAirMassFlow);
+        } else if (thisPIU.UnitType == "AirTerminal:SingleDuct:ParallelPIU:Reheat") {
+            thisPIU.SecAirMassFlow = thisPIU.MaxSecAirMassFlow;
+        }
+    } else if (qdotDelivered1stStageMinFan > zoneLoad) {
+        thisPIU.heatingOperatingMode = HeatOpModeType::HeaterOff;
+        if (thisPIU.UnitType == "AirTerminal:SingleDuct:SeriesPIU:Reheat") {
+            thisPIU.SecAirMassFlow = max(0.0, thisPIU.MinTotAirMassFlow - thisPIU.MinPriAirMassFlow);
+        } else if (thisPIU.UnitType == "AirTerminal:SingleDuct:ParallelPIU:Reheat") {
+            thisPIU.SecAirMassFlow = thisPIU.MinSecAirMassFlow;
+        }
+    }
+}
+
+Real64 CalcVariableSpeedPIUQdotDelivered(EnergyPlusData &state,
+                                         int const PIUNum,   // number of the current PIU being simulated
+                                         int const ZoneNode, // zone node number
+                                         bool const useDAT,
+                                         Real64 const TotAirMassFlow,
+                                         Real64 const fanTurnDown)
+{
```

Comment:
"Helper" function to calculate cooling/heating rate delivered to a zone by a PIU terminal.